File: 3. Arrays/1. Easy/C++/8. Linear search in array.cpp
/*
Linear search in an array
*/

#include <bits/stdc++.h>
using namespace std;


 int linearSearch(int arr[],int n,int num){
    for(int i=0;i<n;i++){
        if(arr[i]==num){
            return i;
        }
    }
    return -1;
}

int main() {
    //Write your code here
    int n;
    cin >> n;
    int arr[n];
    for(int i=0;i<n;i++){
        cin >> arr[i];
    }
    int num;
    cin >> num;
    int ans = linearSearch(arr,n,num);
    cout << ans;
    return 0;
}

File: 3. Arrays/1. Easy/Java/8. Linear search in array.java
/* package codechef; // don't place package name! */

import java.util.*;
import java.lang.*;
import java.io.*;

/* Name of the class has to be "Main" only if the class is public. */

/*
Linear search in an array
*/

class Codechef
{
    
    public static int linearSearch(int arr[],int n,int num){
        for(int i=0;i<n;i++){
            if(arr[i]==num){
                return i;
            }
        }
        return -1;
    }
    
	public static void main (String[] args) throws java.lang.Exception
	{
		// your code goes here
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		int arr[] = new int[n];
		for(int i=0;i<n;i++){
		    arr[i] = sc.nextInt();
 		}
 		int num = sc.nextInt();
 		int ans = linearSearch(arr,n,num);
 		System.out.print(ans);
	}
}
